### DISCUSSION TITLE
i am using google co-lab,I am having an error in pytorch data visualization code, is there anyway to solve the error ?

### DISCUSSION THREAD

**Discussion by tedzz12** [mrdbourke/pytorch-deep-learning#331] [General]
I am  having an error in pytorch data visualization code, here is the code:

import torch
from torch import nn
import matplotlib.pyplot as plt

weight =0.7
bias=0.3
X = torch.arange(0,3,0.02)
Y = weight * X * bias
Training_Set = int(0.8 * len(X))
Training_SetofX = len(X[:Training_Set])
Testing_SetofX = len(X[Training_Set:])
Training_SetofY = len(Y[:Training_Set])
Testing_SetofY = len(Y[Training_Set:])

print(f"Training set of X {len(X[:Training_Set])}")
print(f"Testing set of X {len(X[Training_Set:])}")
print(f"Training set of Y {len(Y[:Training_Set])}")
print(f"Testing set of Y {len(Y[Training_Set:])}")

def plot_predictions(train_data = Training_SetofX,
                     train_label = Training_SetofY,
                     test_data = Testing_SetofX,
                     test_label = Testing_SetofY,
                     predictions = None):
  plt.figure(figsize = (10,7))
  plt.scatter(train_data,train_label,c="b",s=4,label="Training data")
  plt.scatter(test_data,test_label, c="g",s=4,label="Testing data")
if predictions is not None:
  plt.scatter(test_data,predictions,c='r',s=4,label="Predictions")

plt.legend(prop={"size" : 14})

plot_predictions()

here is the error:NameError                                 Traceback (most recent call last)
[<ipython-input-11-8c434664cb88>](https://localhost:8080/#) in <module>
     26   plt.scatter(train_data,train_label,c="b",s=4,label="Training data")
     27   plt.scatter(test_data,test_label, c="g",s=4,label="Testing data")
---> 28 if predictions is not None:
     29   plt.scatter(test_data,predictions,c='r',s=4,label="Predictions")
     30 

NameError: name 'predictions' is not defined

**Comment by pusapatiakhilraju** [comment]
It's working fine for me. Just rerun the cells with correct indentation. Also note that in your code you are storing the lengths of train and test data into the "Training_SetofX", "Training_SetofY", "Testing_SetofX" and "Testing_SetofY" instead of the actual data. Remove the "len()" function to store the actual data.



<img width="1017" alt="Screen Shot 2023-03-08 at 10 30 05 am" src="https://user-images.githubusercontent.com/26441101/223579877-b3f0f779-3178-4300-ad29-daecab6bae5b.png">

**Reply by tedzz12** [reply]
Thank you
